### DISCUSSION TITLE
Get changes from all commits pushed

### DISCUSSION THREAD

**Discussion by murciaelx** [community/community#26215] [Actions]
<p>Hi, thanks for reading.</p>
<p>I was wondering how i can identify the files that have been updated, i know how get this data for the last commit</p>
<p>(</p>
<pre><code>git log -m -1 --name-only --pretty="format:" ${{ github.sha }}
</code></pre>
<p>) but what about if i push 3 commit.</p>
<p>I want to get the filename of all commits pushed on this workflow.</p>
<p>Example:</p>
<p>Here I need to get the modified files of test01, test02 and test03. It is the same push but in different commit</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/1a56401a36e208fa5a72569bbd8dc5f36b295791.jpeg" data-download-href="/uploads/short-url/3KZcHudjKLY90gWQ4DPLjnwCSY1.jpg?dl=1" title="tt.JPG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/30163424/181096098-0756ddae-6c3a-4020-821d-d9530cf54a6b.jpeg" alt="tt.JPG" data-base62-sha1="3KZcHudjKLY90gWQ4DPLjnwCSY1" width="690" height="131" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/1a56401a36e208fa5a72569bbd8dc5f36b295791_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">tt.JPG</span><span class="informations">990×189 15 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks</p>

**Comment by logankilpatrick** [comment]
<p>Hey, I think your best bet is to do the following: <a href="https://stackoverflow.com/questions/1552340/how-to-list-only-the-file-names-that-changed-between-two-commits" rel="nofollow noopener">https://stackoverflow.com/questions/1552340/how-to-list-only-the-file-names-that-changed-between-two-commits</a></p>
<p>I do not think Git looks at the changes in files the same way you are trying to, meaning that you can’t compare across pushes and need to compare across commits. </p>

**Comment by murciaelx** [comment, marked as answer]
<p>Thanks for your reply <a class="mention" href="/logankilpatrick">@logankilpatrick</a>! </p>
<p>After battling I have found a useful solution.</p>
<pre><code>- uses: actions/checkout@v2
        with:
          fetch-depth: 0

git diff --name-only ${{ github.event.before }} ${{ github.sha }}
</code></pre>
